luigi/contrib/gcs.py

```diff
@@ -64,6 +67,29 @@
 GCS_BATCH_URI = 'https://storage.googleapis.com/batch/storage/v1'
 
 
+# Retry configuration. For more details, see https://tenacity.readthedocs.io/en/latest/
+def custom_retry(retry_state):
+    """custom_retry returns true if the outcome produced the function is retryable,
+    false otherwise.
+    """
+    try:
+        retry_state.outcome.result()
```

Comment:
I haven't used tenacity before, but I think this will return the actual result of the wrapped call, whereas you below return a boolean.

I think it can be expressed easier with retry_if_exception with a custom predicate, e.g.

```
def is_error_5xx(err: Exception):
    return isinstance(err, errors.HttpError) and e.resp >= 500

gcs_retry = retry(retry=retry_if_exception(is_error_5xx),
    ...

```

Reply:
I applied this suggestion 8d88eb0. Thanks!